ocdskingfisherprocess/checks.py

```diff
@@ -43,37 +45,92 @@ def process_all_files(self):
         # Checks with schema V1.1
         if self.collection.check_older_data_with_schema_version_1_1:
 
-            self.process_all_files_releases_with_override_schema_version_1_1()
+            self.process_releases_with_override_schema_version_1_1(
+                self.database.get_releases_to_check(self.collection.database_id, override_schema_version="1.1")
+            )
 
             # Early return?
             if self.run_until_timestamp and self.run_until_timestamp < datetime.datetime.utcnow().timestamp():
                 return
 
-            self.process_all_files_records_with_override_schema_version_1_1()
+            self.process_records_with_override_schema_version_1_1(
+                self.database.get_records_to_check(self.collection.database_id, override_schema_version="1.1")
+            )
+
+    def process_file_item_id(self, collection_file_item_id):
+
+        self.logger.info('process_file_item_id called for collection file item id ' + str(collection_file_item_id))
+
+        # Is deleted?
+        if self.collection.deleted_at:
+            return
+
+        # Normal Checks
+        if self.collection.check_data:
+
+            s = sa.sql.select([self.database.release_table]) \
+                .where(self.database.release_table.c.collection_file_item_id == collection_file_item_id)
+            with self.database.get_engine().begin() as connection:
+                releases = connection.execute(s)
+            self.process_releases(releases)
```

Comment:
As a readability thing I would have the `if self.collection.check_data` here and at `process_record` rather than copy the this whole section.

Reply:
I'm not sure which code you mean?
I think for speed it's good to have "if self.collection.check_data:" as early as possible in the process - 2/3 of our collections now have checks turned off, so in those cases an early return as fast as possible is good!

Reply:
I did not consider the speed issue.

Maybe a guard at the top with 
``` python
if not (self.collection.check_data or self.collection.check_older_data_with_schema_version_1_1):
   return
```

then 

if statements around  `self._process_releases(releases)` and `self._process_records(records)`
and else being '_process_releases_with_override_schema_version_1_1(releases)' and  '_process_records_with_override_schema_version_1_1(records)' respectively. 

I think I would find that more readable as I naturally tried to see the differences in the two sections and took me while to realise there was not any apart from what function they called.


However I do not care that much as its just a readability thing.

Reply:
Can we leave it like it is?  process_all_files is structured like it is because it only gets the data that actually needs checking, so each section is different. In the future we may want to do the same to process_file_item_id, and so having it laid out the same way is good for that.